lib/format_parser.rb

```diff
@@ -58,7 +73,22 @@ def self.parse(io)
         # and examine the file more closely.
       end
     end
-    nil # Nothing matched
+    # Return the array of results if something matched or nil otherwise.
+    results.empty? ? nil : results
+  end
+
+  def self.parse_config(**opts, &proc)
```

Comment:
OpenStruct is known to be slow, besides - see my remark up the code about overinflating this interface. I think just having kwargs explicitly declared in the method signature with sane defaults will do.